atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/CellLoadingBatcher.java

```diff
@@ -20,91 +20,71 @@
-                rebatchingManyRowsWarningCallback);
-    }
-
-    List<List<Cell>> partitionIntoBatches(Collection<Cell> cellsToPartition) {
-        int multigetMultisliceBatchLimit = crossColumnLoadBatchLimitSupplier.getAsInt();
-        int singleQueryLoadBatchLimit = singleQueryLoadBatchLimitSupplier.getAsInt();
+    List<List<Cell>> partitionIntoBatches(Collection<Cell> cellsToPartition,
+            IntConsumer rebatchingManyRowsWarningCallback) {
+        CassandraCellLoadingConfig config = loadingConfigSupplier.get();
 
         ListMultimap<byte[], Cell> cellsByColumn = indexCellsByColumnName(cellsToPartition);
 
         List<List<Cell>> batches = Lists.newArrayList();
         List<Cell> cellsForCrossColumnBatching = Lists.newArrayList();
         for (Map.Entry<byte[], List<Cell>> cellColumnPair : Multimaps.asMap(cellsByColumn).entrySet()) {
-            if (shouldExplicitlyAllocateBatchToColumn(multigetMultisliceBatchLimit, cellColumnPair.getValue())) {
-                batches.addAll(
-                        partitionBySingleQueryLoadBatchLimit(cellColumnPair.getValue(), singleQueryLoadBatchLimit));
+            if (shouldExplicitlyAllocateBatchToColumn(config, cellColumnPair.getValue())) {
+                batches.addAll(partitionBySingleQueryLoadBatchLimit(
+                        cellColumnPair.getValue(), config, rebatchingManyRowsWarningCallback));
             } else {
                 cellsForCrossColumnBatching.addAll(cellColumnPair.getValue());
             }
         }
-        batches.addAll(Lists.partition(cellsForCrossColumnBatching, multigetMultisliceBatchLimit));
+        batches.addAll(Lists.partition(cellsForCrossColumnBatching, config.crossColumnLoadBatchLimit()));
 
         return batches;
     }
 
-    private List<List<Cell>> partitionBySingleQueryLoadBatchLimit(List<Cell> cells, int singleQueryLoadBatchLimit) {
-        if (cells.size() > singleQueryLoadBatchLimit) {
+    private List<List<Cell>> partitionBySingleQueryLoadBatchLimit(
+            List<Cell> cells, CassandraCellLoadingConfig config, IntConsumer rebatchingManyRowsWarningCallback) {
+        if (cells.size() > config.singleQueryLoadBatchLimit()) {
             rebatchingManyRowsWarningCallback.accept(cells.size());
-            return Lists.partition(cells, singleQueryLoadBatchLimit);
+            return Lists.partition(cells, config.singleQueryLoadBatchLimit());
         }
         return ImmutableList.of(cells);
     }
 
-    private static boolean shouldExplicitlyAllocateBatchToColumn(int multigetMultisliceBatchLimit, List<Cell> cells) {
-        return cells.size() > multigetMultisliceBatchLimit;
+    private static boolean shouldExplicitlyAllocateBatchToColumn(CassandraCellLoadingConfig config, List<Cell> cells) {
+        return cells.size() > config.crossColumnLoadBatchLimit();
     }
 
     private static ListMultimap<byte[], Cell> indexCellsByColumnName(Collection<Cell> cells) {
```

Comment:
We can lose the comment below

Reply:
We can - though I think it should be included, as it looks like something people might instinctively change to the simpler method without realising the implications